src/SignalR/server/Core/src/HubLifetimeManager.cs

```diff
@@ -131,4 +134,41 @@ public abstract class HubLifetimeManager<THub> where THub : Hub
     /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None" />.</param>
     /// <returns>A <see cref="Task"/> that represents the asynchronous remove.</returns>
     public abstract Task RemoveFromGroupAsync(string connectionId, string groupName, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Sends an invocation message to the specified connection and waits for a response.
+    /// </summary>
+    /// <typeparam name="T">The type of the response expected.</typeparam>
+    /// <param name="connectionId">The connection ID.</param>
+    /// <param name="methodName">The invocation method name.</param>
+    /// <param name="args">The invocation arguments.</param>
+    /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None" />.</param>
+    /// <returns>The response from the connection.</returns>
+    public virtual Task<T> InvokeConnectionAsync<T>(string connectionId, string methodName, object?[] args, CancellationToken cancellationToken = default)
```

Comment:
I wonder if we need a SupportsClientResults bool so we can throw a nicer error when implementations don't support it. We can check and throw in the ISingleClientProxy

Reply:
Not sure why a new API is needed. I updated the error message to:
$"{GetType().Name} does not support client return values." so it's clear what the problem is.